### PR TITLE
jdk17+: remove Unsafe.defineAnonymousClass()

### PR DESCRIPTION
**Pull request by keithc-ca**
The method `defineAnonymousClass()` was removed from both `jdk.internal.misc.Unsafe` and `sun.misc.Unsafe` by:
* 8243287: Removal of Unsafe::defineAnonymousClass

### CODE REVIEW

**Comment by keithc-ca** [comment]
Jenkins test sanity,extended zlinux,win jdk11,jdk17

File: runtime/jcl/uma/sun_misc_Unsafe_exports.xml [hunk 1]
```diff
@@ -37,7 +37,9 @@
 		<export name="Java_sun_misc_Unsafe_throwException" />
 		<export name="Java_sun_misc_Unsafe_tryMonitorEnter" />
 		<export name="Java_sun_misc_Unsafe_unpark" />
-		<export name="Java_sun_misc_Unsafe_defineAnonymousClass" />
+		<export name="Java_sun_misc_Unsafe_defineAnonymousClass">
+			<exclude-if condition="spec.java17" />
```
Comment (gacholio) [inline runtime/jcl/uma/sun_misc_Unsafe_exports.xml:41]: Is `spec.java17` going to be set for future versions of java?

Comment (keithc-ca) [reply runtime/jcl/uma/sun_misc_Unsafe_exports.xml:41]: Yes, `spec.java17` is equivalent to `JAVA_SPEC_VERSION >= 17`.

**Comment by keithc-ca** [comment]
Apparently some tests need to be updated.

**Comment by hangshao0** [comment]
I think those Unsafe.defineAnonymousClass tests can be removed on Java 17+ to unblock this, as the API itself is gone. I can then later add the equivalent back in https://github.com/eclipse-openj9/openj9/issues/13529

**Comment by DanHeidinga** [comment]
@keithc-ca @hangshao0 what's the next step on this PR?  It's currently in the 0.28 (Java 17) plan - does it need to be resolved for the release?

**Comment by keithc-ca** [comment]
I was just finishing up the changes required for tests (that can no longer use the remove API); I plan to push those changes here shortly. I have cherry-picked changes ready for the 0.28.0 release should we decide to include it there.

**Comment by keithc-ca** [comment]
A new commit has been added with the test-related changes and the original commit has been rebased on that.

**Comment by keithc-ca** [comment]
Jenkins test sanity,extended zlinux,win jdk8,jdk11,jdk17

**Comment by pshipton** [comment]
lgtm

**Pull request merged**